File: src/ts/hammingWeight.ts
export function hammingWeight (n: number): number {
  let count = 0
  while (n !== 0) {
    n = n & (n - 1)
    count += 1
  }

  return count
}

export function hammingWeight1 (n: number): number {
  let count = 0
  while (n !== 0) {
    n &= n - 1
    count += 1
  }

  return count
}

File: src/rs/hammingWeight.rs
#[allow(dead_code)]
pub struct HammingWeight {}

#[allow(dead_code, non_snake_case)]
impl HammingWeight {
    pub fn hammingWeight(n: u32) -> i32 {
        let mut u = n;
        let mut count = 0;
        while u != 0 {
            u = u & (u - 1);
            count += 1;
        }
        count
    }

    pub fn hammingWeight1(n: u32) -> i32 {
        let mut u = n;
        let mut count = 0;
        while u != 0 {
            u &= u - 1;
            count += 1;
        }
        count
    }
}
